core/src/main/scala/com/mesosphere/usi/core/javadsl/SchedulerAdapter.scala

```diff
@@ -0,0 +1,84 @@
+package com.mesosphere.usi.core.javadsl
+
+import java.util.Optional
+import java.util.concurrent.{CompletableFuture, CompletionStage}
+
+import akka.{Done, NotUsed}
+import akka.stream.{Materializer, OverflowStrategy, QueueOfferResult, scaladsl}
+import akka.stream.javadsl.{Flow, Sink, SinkQueueWithCancel, Source, SourceQueueWithComplete}
+import com.mesosphere.usi.core.Scheduler.{SpecInput, StateOutput}
+import com.mesosphere.usi.core.scaladsl.{SchedulerAdapter => ScalaSchedulerAdapter}
+import com.mesosphere.usi.core.models.{SpecUpdated, SpecsSnapshot, StateEvent, StateSnapshot}
+
+import scala.compat.java8.FutureConverters._
+import akka.japi.tuple.Tuple3
+import akka.japi.Pair
+
+class SchedulerAdapter(schedulerFlow: scaladsl.Flow[SpecInput, StateOutput, NotUsed], mat: Materializer) {
+
+  private implicit val materializer: Materializer = mat
+
+  private val delegate = new ScalaSchedulerAdapter(schedulerFlow)
+
+  /**
+    * Represents the scheduler as a Sink and Source.
+    *
+    * This method will materialize the scheduler first, then Sink and Source can be materialized independently.
+    *
+    * @param specsSnapshot Snapshot of the current specs
+    * @return Snapshot of the current state, as well as Source which produces StateEvents and Sink which accepts SpecEvents
+    */
+  def asSourceAndSink(specsSnapshot: SpecsSnapshot = SpecsSnapshot.empty)
+    : Tuple3[CompletableFuture[StateSnapshot], Source[StateEvent, Any], Sink[SpecUpdated, Any]] = {
+    val (snap, source, sink) = delegate.asSourceAndSink(specsSnapshot)
+    Tuple3(snap.toJava.toCompletableFuture, source.asJava, sink.asJava)
```

Comment:
I wonder if the tuple should be a future since one should start pushing `SpecUpdated` and pulling `StateEvent` *after* the `StateSnapshot` arrived.

Reply:
Hmm, I'd recommend using `NotUsed` over `Any`.

Reply:
I don't think it's a problem since all events that arrive before `StateSnapshot`will be backpressured.